test/functional/sessions_tests.js

```diff
@@ -27,15 +27,19 @@ describe('Sessions', function() {
   });
 
   it('should send endSessions for multiple sessions', {
-    metadata: { requires: { topology: ['single'], mongodb: '>3.6.0-rc0' } },
+    metadata: {
+      requires: { topology: ['single'], mongodb: '>3.6.0-rc0' },
+      // Skipping session leak tests b/c these are explicit sessions
+      sessions: { skipLeakTests: true }
+    },
     test: function(done) {
       var client = this.configuration.newClient({ w: 1 }, { poolSize: 1, auto_reconnect: false });
       client.connect((err, client) => {
         let sessions = [client.startSession(), client.startSession()].map(s => s.id);
 
         client.close(err => {
           expect(err).to.not.exist;
-          expect(test.commands.started).to.have.length(1);
+          expect(test.commands.started).to.have.length(2);
```

Comment:
what is the second command that's started here?

Reply:
if I am correct, there are two duplicate calls to `endSession` happening. I am not sure why. I can investigate more.